jsonpatcher/jsonpatcher.go

```diff
@@ -220,21 +232,21 @@ func saveEvent(id core.InstanceID, prev []byte, curr []byte) (*operation, error)
 
 func deleteEvent(id core.InstanceID) (*operation, error) {
 	return &operation{
-		Type:       delete,
+		Type:       del,
 		InstanceID: id,
 		JSONPatch:  nil,
 	}, nil
 }
 
 type patchEvent struct {
-	Timestamp      time.Time
+	Timestamp      int64
```

Comment:
@sanderpick FYI this change breaks the backward compatibility with existing logs/dbs...
Not sure if you have an intent to promise this by the moment. 

Let me know, I can propose PR to support the old format

Reply:
Hrm, okay. Let's see... what do you propose? To add some context, `time.Time` isn't cbor-serializable since it doesn't export any fields.

Reply:
You can try to cbor-serialize the old-format patchEvent(with time.Time) and then try to deserialize it into the new format(with int64) or vice versa. You will get unmarshal error.
Here is the proof to test: 
https://gist.github.com/requilence/d80e0eb1e346a8b40403e8593163aabf

Reply:
Added a comment to your gist: https://gist.github.com/requilence/d80e0eb1e346a8b40403e8593163aabf#gistcomment-3438857

Reply:
@sanderpick thanks, this solution you've proposed seems ok, btw older clients will not be able to read the data created with the newer version.

Personally, I don't like the way these cbor-serialized structs become no-upgradable at all. In case of any struct amend, you need to preserve the old struct and still break the non-upgraded clients that read the same data.

On the other hand, PB-encoded ipld nodes look more solid as it provides the way to add new fields and remove the old ones as far as you don't reuse the same indexes. It also ignores the extra fields, which means compatible with older clients. Maybe you can provide some context why did you prefer cbor?

Anyway, it may be also useful to add some comments on all cbor-encoded structs that it shouldn't be changed

Reply:
Sorry for the delay. Coming up for air :) Agreed, the upgrade-ability is poor. IMO, CBOR is much nicer to work with compared to proto nodes in every other way. In any case, it would be an undertaking to migrate to proto nodes. Ideally, we could use any kind of `format.Node` at the `net` layer.

We could introduce a `Version` to the `core.Event` to make the handling a little cleaner. Still, older nodes are out of luck on this one.

Reply:
I ended up going with a simpler solution: https://github.com/textileio/go-threads/pull/439